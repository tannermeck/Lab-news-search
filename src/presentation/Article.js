import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

export const Article = ({author, title, description}) => {
    return (
    <>
        <p aria-label="author-tag">Author: {author}</p>
        <p>Title: {title}</p>
        <p>Description: {description}</p>
    </>
    )
}

Article.propTypes = {
    author: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
}