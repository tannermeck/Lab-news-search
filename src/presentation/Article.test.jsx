import React from 'react';
import { render } from '@testing-library/react'
import { Article } from './Article';

describe('should render a list of articles', () => {
    it('it should render a list of authors, titles, and descriptions', () => {
        const { asFragment } = render(
        <Article title='bitcoin' author='Elon Musk' description='Bitcoin soars' />);
        expect(asFragment()).toMatchSnapshot()
    })
})