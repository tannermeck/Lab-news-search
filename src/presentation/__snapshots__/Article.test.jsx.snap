// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of articles it should render a list of authors, titles, and descriptions 1`] = `
<DocumentFragment>
  <p
    aria-label="author-tag"
  >
    Author: Elon Musk
  </p>
  <p>
    Title: bitcoin
  </p>
  <p>
    Description: Bitcoin soars
  </p>
</DocumentFragment>
`;
